docs/custom_plugin.md

```diff
@@ -0,0 +1,429 @@
+
+            # Flag which decides if an explicit include from the
+            # command line is requiremd to enable this requirement.
+            requires_explicit_include=False,
+        )
+
+        # The dynamic_arg_name is used in defining the dynamic
+        # argument `ReadmeCheck-no-Readme`.
+        self.dynamic_arg_name: str = "no"
+
+        # Flag deciding if the requirement is enabled by default.
+        self.enabled_by_default: bool = True
+
+    @override
+    def GetDynamicArgDefinitions(
+            self,
+            argument_separator: str) -> dict[str, TypeDefinitionItemType]:
+        """Get the definitions for the arguments to this requirement."""
+
+        # Defines a dynamic argument `ReadmeCheck-no-Readme` which if enabled
+        # allows the requirement to pass if no README file is available.
+        return {
+            f"{self.dynamic_arg_name}{argument_separator}{self.name}": (
+                bool,
+                typer.Option(
+                    default=False,
+                    help="Allow missing README.",
+                ),
+            ),
+        }
+
+    @override
+    def _EvaluateImpl(
+        self,
+        query_data: dict[str, Any],
+        requirement_args: dict[str, Any],
+    ) -> Requirement.EvaluateImplResult:
+        """Implementation of the reqiurement evaluation.
+
+        query_data: The data obtained from the query class (detailed next).
+        requirement_args: Reqiurement specific dynamic arguments.
+        """
+
+        # Flag to enable check for the file.
+        check_for_existence: bool = self.enabled_by_default
+
+        # Check if the dynamic argument is provided,
+        # which toggles the check flag
+        if requirement_args.get(self.dynamic_arg_name, False):
+            check_for_existence = not check_for_existence
+
+        # If check flag is enabled, we perform the check.
+        # The repository directory is in `query_data[repo_dir]`
+        # thanks to the Query.py file.
+        if check_for_existence:
+
+            # Get full file path in temporary repo directory and
+            # check if it exists.
+            # This checks both upper case and lower case variants
+            full_file_path = Path(query_data["repo_dir"].name) / "README.md"
+
+            # If the README file exists, return success.
+            if full_file_path.exists():
+                return Requirement.EvaluateImplResult(
+                    EvaluateResult.Success,
+                    "README found inside the repository",
+                )
+
+            # Else return error.
+            return Requirement.EvaluateImplResult(
+                EvaluateResult.Error,
+                f"No {self.filename} file found.",
+                provide_resolution=True,
+                provide_rationale=True,
+            )
+
+        # Requirement not enabled, so return DoesNotApply
+        return Requirement.EvaluateImplResult(EvaluateResult.DoesNotApply,
```

Comment:
See my comment above, as we got it wrong again :) 

There are 3 states:

- Default: Ensure that `README.md` exists.
- `ReadmeCheck-no-Readme`: Ensure that `README.md` does not exist.
- `--exclude Readme`: `EvaluateResult.DoesNotApply`.

This code needs to add the path that enforces `README.md` does not exist.